great_tables/_tbl_data.py

```diff
@@ -759,7 +759,7 @@ def _(df: PyArrowTable, x: Any) -> bool:
     import pyarrow as pa
 
     arr = pa.array([x])
-    return arr.is_null().to_pylist()[0] or arr.is_nan().to_pylist()[0]
+    return arr.is_null(nan_is_null=True).to_pylist()[0]
```

Comment:
Reason for this change is that `.is_nan` raises an exception for non-numeric arrays